File: ScorpioConversion/Sample/go/src/scov/TableSpawn.go
package scov
import (
    "scorpioproto"
)


// TableSpawn 类名
type TableSpawn struct {
    count int
    dataArray map[int32]*DataSpawn
}

// Initialize 初始化
func (table *TableSpawn) Initialize(fileName string, reader scorpioproto.IScorpioReader) {
    if table.dataArray == nil {
		table.dataArray = map[int32]*DataSpawn{}
	}
    iRow := scorpioproto.TableUtilReadHead(reader, fileName, "484cdae7d179982f1c7868078204d81d");
    for i := 0; i < iRow; i++ {
        pData := DataSpawnRead(fileName, reader)
        if table.Contains(pData.ID()) {
            table.dataArray[pData.ID()].Set(pData)
        } else {
            table.dataArray[pData.ID()] = pData;
        }
    }
    table.count = len(table.dataArray)
}

// Contains 是否包含数据
func (table *TableSpawn) Contains(ID int32) bool {
    if _, ok := table.dataArray[ID]; ok {
        return true
    }
    return false
}

// GetValue 获取数据
func (table *TableSpawn) GetValue(ID int32) *DataSpawn {
    if table.Contains(ID) {
        return table.dataArray[ID];
    }
    //TableUtilWarning("DataSpawn key is not exist " + ID);
    return nil;
}

// Datas 所有数据
func (table *TableSpawn) Datas() map[int32]*DataSpawn {
    return table.dataArray
}

// Count 数量
func (table *TableSpawn) Count() int {
    return table.count
}

File: Backup/ts/src/Code/TableSpawn.ts
import { IScorpioReader } from '../ScorpioProto/Commons/IScorpioReader'
import { ScorpioUtil } from '../ScorpioProto/Commons/ScorpioUtil'
import { TableUtil } from '../ScorpioProto/Table/TableUtil'
import { IData } from '../ScorpioProto/Table/IData'
import { ITable } from '../ScorpioProto/Table/ITable'
import { DataSpawn } from './DataSpawn';
export class TableSpawn implements ITable {
    private m_count:number = 0;
    private m_dataArray:{[key:number]:DataSpawn} = {};
    public Initialize(fileName:string, reader:IScorpioReader):TableSpawn {
        let iRow = TableUtil.ReadHead(reader, fileName, "484cdae7d179982f1c7868078204d81d");
        for (let i = 0; i < iRow; ++i) {
            let pData = DataSpawn.Read(fileName, reader);
            if (this.Contains(pData.ID())) {
                this.m_dataArray[pData.ID()].Set(pData);
            } else {
                this.m_dataArray[pData.ID()] = pData;
            }
        }
        this.m_count = Object.getOwnPropertyNames(this.m_dataArray).length;
        return this;
    }
    public GetValue(ID:number):DataSpawn|null {
        if (this.Contains(ID)) return this.m_dataArray[ID];
        TableUtil.Warning("DataSpawn key is not exist " + ID);
        return null;
    }
    public Contains(ID:number):boolean {
        return this.m_dataArray[ID] != null;
    }
    public Datas():{[key:number]:DataSpawn} {
        return this.m_dataArray;
    }
    public GetValueObject(ID:any):IData|null {
        return this.GetValue(ID as number);
    }
    public ContainsObject(ID:any):boolean {
        return this.Contains(ID as number);
    }
    public GetDatas():any {
        return this.Datas();
    }
    public Count():number {
        return this.m_count;
    }
}
